library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
So we already have `has_quote`, and it's called `to_pexpr`?   Great.

> Adding a type annotation around (pexpr.subst (pexpr.subst ...)) would not help.

Why not?  The error that we want to prevent here is that the user returns `nat.zero` instead of `pexpr.subst ...`.  We want to make sure that the type of `pexpr.subst ...` is indeed `α = pexpr` and not something else.